pebbles/drivers/provisioning/openshift_driver.py

```diff
@@ -0,0 +1,666 @@
+import base64
+import json
+import time
+import uuid
+from pprint import pprint
+import requests
+# noinspection PyUnresolvedReferences
+from six.moves.urllib.parse import urlparse, parse_qs
+
+from pebbles.client import PBClient
+from pebbles.drivers.provisioning import base_driver
+
+# maximum time to wait for pod creation before failing
+MAX_POD_SPAWN_WAIT_TIME_SEC = 600
+# maximum time to wait for pod (down) scaling
+MAX_POD_SCALE_WAIT_TIME_SEC = 60
+
+
+class OpenShiftClient(object):
+    def __init__(self, base_url, subdomain, user, password):
+        if base_url[-1] == '/':
+            base_url = base_url[:-1]
+        self.base_url = base_url
+        self.subdomain = subdomain
+        self.oapi_base_url = base_url + '/oapi/v1'
+        self.kube_base_url = base_url + '/api/v1'
+        self.user = user
+        self.password = password
+
+        # token_data caches the token to access the API. See _request_token() for details.
+        self.token_data = None
+        self._session = requests.session()
+
+    @staticmethod
+    def make_base_kube_object(kind, name=None):
+        return dict(
+            kind=kind,
+            apiVersion="v1",
+            metadata=dict(
+                name=name
+            )
+        )
+
+    @staticmethod
+    def print_resp(resp):
+        if resp.ok:
+            print('success: %s' % resp.status_code)
+            pprint(resp.json())
+        else:
+            print('error in response: %s %s %s' % (resp.status_code, resp.reason, resp.text))
+
+    def _request_token(self, current_ts=None):
```

Comment:
looks fairly simple, but untested. since you're using responses or whatevs could you test this as well?